### DISCUSSION TITLE
Commands and Extensions

### DISCUSSION THREAD

**Discussion by mocenigo** [zed-industries/zed#18061] [Feature Requests]
Hello, how do i create commands in the settings.json file, for instance, to call a specific external tool?
I guess that then adding a keybinding would be easy.
But,  is there already a mechanism to do all of this into an extension? (The same question was asked in #17226)
It would be nice to have a tutorial with one example.

On a related note, hoow can one add snippets into an extension, and add visual clues to the editor on how to make the corresponding snippets panel appear in a dock? This could also go into that tutorial

**Comment by AlvaroParker** [comment]
You can use tasks to run commands using the integrated terminal. See https://zed.dev/docs/tasks

**Comment by mocenigo** [comment]
Solved for Skim

I disable forwardSearch on save, and define the following command

```
[
  {
    "label": "Forward Search",
    "command": "str=$(grep -e \"%.*TEX root = \" -m 1 -n $ZED_FILE); if [ \"$str\" = \"\" ]; then str=$ZED_STEM; else str=$(echo $str | sed \"s/[0-9]*:%.*TEX root = \\(.*\\).tex/\\1/\"); fi; echo $str; /Applications/Skim.app/Contents/SharedSupport/displayline -r -b $ZED_ROW $ZED_DIRNAME/$str.pdf $ZED_FILE",
    "allow_concurrent_runs": false,
    "reveal": "never ",
    "hide": "always",
  },
]

```

It sucks that JSON does not allow line split literals. 

Note that this allows a file to be opened and previewed from an included one if the included file contains a line like 
`% TeX root = main-file-name.tex`
or
`% !TeX root = main-file-name.tex`
(with or without a space after the %).

The Skim displaylike allows to give also a tex file name as the parameter AFTER the pdf file and this version allows to jump also to the right place FROM an included file, as long as there is a commented line containint 
"TEX root = " followed by the file name (also, respect the spaces between these words and after the "=", as in TeXshop and similar programs.

I guess this should be included in the latex-lsp extension with an option to install the Skim.app command.

 Roberto

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
